src/helpers.c

```diff
@@ -72,17 +71,22 @@ int rm_staging_dir_contents(const char *rel_path)
 		return -1;
 	}
 
-	while ((ret = readdir_r(dir, &entry, &result)) == 0) {
-		if (result == NULL) {
+	errno = 0;
+	while(true) {
+		entry = readdir(dir);
+		if (!entry) {
+			if (errno) {
+				ret = errno;
```

Comment:
If errno (for some reason) isn't set then we don't set ret and return an uninitialized int. We should initialize it to -1 and also set it to 0 after the last free(filename)  - in case no files were removed or it's an empty directory.